### PR TITLE
Bug 1178229 - Use new syntax to define the wallpaper module r=eragonj

### PR DESCRIPTION
**Pull request by crh0716**
https://bugzilla.mozilla.org/show_bug.cgi?id=1178229

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=cfdfd02efbb2abf82a64bddfe174fd49d39a52c1

File: apps/settings/js/panels/display/wallpaper.js [hunk 1]
```diff
@@ -14,133 +14,135 @@ define(function(require) {
...
+   * Watch the value of wallpaper.image from settings and change wallpaperSrc.
+   *
+   * @access private
+   * @memberOf Wallpaper
+   */
+  Wallpaper.prototype._watchWallpaperChange = function() {
+    SettingsListener.observe(this.WALLPAPER_KEY, '',
+      function onHomescreenchange(value) {
+        this._wallpaperSrc = this.wallpaperURL.set(value);
+    }.bind(this));
```
Comment (EragonJ) [inline apps/settings/js/panels/display/wallpaper.js]: why not use arrow function

**Comment by EragonJ** [comment]
Why not use arrow function :) ?

File: apps/settings/js/panels/display/wallpaper.js [hunk 2]
```diff
@@ -14,133 +14,135 @@ define(function(require) {
...
+        includeLocked: (secret !== null),
+        // XXX: This will not work with Desktop Fx / Simulator.
+        width: Math.ceil(window.screen.width * window.devicePixelRatio),
+        height: Math.ceil(window.screen.height * window.devicePixelRatio)
       }
-    },
+    });
+    mozActivity.onsuccess = function() {
+      this._onPickSuccess(mozActivity.result.blob, secret);
+    }.bind(this);
```
Comment (EragonJ) [inline apps/settings/js/panels/display/wallpaper.js]: why not use arrow function :) ?

File: apps/settings/js/panels/display/wallpaper.js [hunk 3]
```diff
@@ -14,133 +14,135 @@ define(function(require) {
...
+   */
+  Wallpaper.prototype._onPickSuccess = function(blob, secret) {
+    if (!blob) {
+      return;
+    }
+    if (blob.type.split('/')[1] === ForwardLock.mimeSubtype) {
+      // If this is a locked image from the locked content app, unlock it
+      ForwardLock.unlockBlob(secret, blob, function(unlocked) {
+        this._setWallpaper(unlocked);
+      }.bind(this));
```
Comment (EragonJ) [inline apps/settings/js/panels/display/wallpaper.js]: same here !

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d13e01ba11fb821db2c94be17e2dad3c08836441

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=29de0ad39f3a0c6a32bba276b81a0715fe303df8

**Pull request merged**